test/e2e/storage/persistent_volumes-local.go

```diff
@@ -420,9 +421,158 @@ var _ = utils.SIGDescribe("PersistentVolumes-local ", func() {
 		})
 	})
 
-	// TODO: add stress test that creates many pods in parallel across multiple nodes
+	Context("Stress with local volume provisioner [Serial]", func() {
+		var testVols [][]string
+
+		const (
+			volsPerNode = 10 // Make this non-divisable by volsPerPod to increase changes of partial binding failure
+			volsPerPod  = 3
+			podsFactor  = 5
+		)
+
+		BeforeEach(func() {
+			setupStorageClass(config, &waitMode)
+			setupLocalVolumeProvisioner(config)
+
+			testVols = [][]string{}
+			for i, node := range config.nodes {
+				By(fmt.Sprintf("Setting up local volumes on node %q", node.Name))
+				paths := []string{}
+				for j := 0; j < volsPerNode; j++ {
+					volumePath := path.Join(config.discoveryDir, fmt.Sprintf("vol-%v", string(uuid.NewUUID())))
+					setupLocalVolumeProvisionerMountPoint(config, volumePath, &config.nodes[i])
+					paths = append(paths, volumePath)
+				}
+				testVols = append(testVols, paths)
+			}
+
+			By("Starting the local volume provisioner")
+			createProvisionerDaemonset(config)
+		})
+
+		AfterEach(func() {
+			By("Deleting provisioner daemonset")
+			deleteProvisionerDaemonset(config)
+
+			for i, paths := range testVols {
+				for _, volumePath := range paths {
+					cleanupLocalVolumeProvisionerMountPoint(config, volumePath, &config.nodes[i])
+				}
+			}
+			cleanupLocalVolumeProvisioner(config)
+			cleanupStorageClass(config)
+		})
+
+		It("should use be able to process many pods and reuse local volumes", func() {
+			var (
+				podsLock sync.Mutex
+				// Have one extra pod pending
+				numConcurrentPods = volsPerNode/volsPerPod*len(config.nodes) + 1
+				totalPods         = numConcurrentPods * podsFactor
+				numCreated        = 0
+				numFinished       = 0
+				pods              = map[string]*v1.Pod{}
+			)
+
+			// Create pods gradually instead of all at once because scheduler has
+			// exponential backoff
+			// TODO: this is still a bit slow because of the provisioner polling period
+			By(fmt.Sprintf("Creating %v pods periodically", numConcurrentPods))
+			stop := make(chan struct{})
+			go wait.Until(func() {
+				if numCreated >= totalPods {
+					// Created all the pods for the test
+					return
+				}
+
+				podsLock.Lock()
```

Comment:
nit: Lock this at top of function, to prevent this case as much as possible:

1. `Until()` triggers an execution of the func
1. `defer` closes the channel, grabs the lock before the func does.

Reply:
done